src/main/java/de/exercises/pithonAPI/PiThonAPI.java

```diff
@@ -0,0 +1,56 @@
+package de.exercises.pithonAPI;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class PiThonAPI {
+    public static final String PITHON_URL = "http://localhost:5000/api";
+
+    public static void main(String[] args) {
+        if(args[0] == null) {
+            System.out.println("argument(s) missing: number (pi, e, sqrt2) and amount(optional)");
+            return;
+        }
+        String number = args[0];
+        int amount;
+        if(args[1] != null) {
+            try {
+                amount = Integer.parseInt(args[1]);
+            } catch (NumberFormatException e) {
+                System.out.println("Unable to read amount (second argument)");
+                return;
+            }
+        } else {
+            amount = 10;
+        }
+        System.out.println(getDigits(number, amount));
+    }
+
+    public static String getDigits(String number, int amount) {
+        try {
+            String params = "?number=" + number + "&amount=" + amount + "&index=0";
+            URL url = new URL(PITHON_URL + params);
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
+            con.setRequestMethod("GET");
```

Comment:
An interesting option here would be to also add basic authentication. Just to get a feel for how to do it in Java.